tests/phpunit/integration/Modules/Tag_ManagerTest.php

```diff
@@ -36,11 +36,150 @@ public function test_register() {
 		);
 	}
 
-	public function test_is_connected() {
+	public function test_register_wp_amp() {
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$mock_context = $this->getMockBuilder( 'MockClass' )->setMethods( array( 'is_amp', 'input' ) )->getMock();
+		$mock_context->method( 'input' )->will( $this->returnValue( $context->input() ) );
+		$mock_context->method( 'is_amp' )->will( $this->returnValue( true ) );
+
+		$tagmanager = new Tag_Manager( $context );
+		$this->force_set_property( $tagmanager, 'context', $mock_context );
+
+		remove_all_actions( 'wp' );
+		$tagmanager->register();
+
+		remove_all_actions( 'amp_print_analytics' );
+		remove_all_actions( 'wp_footer' );
+		remove_all_actions( 'amp_post_template_footer' );
+		remove_all_filters( 'amp_post_template_data' );
+
+		do_action( 'wp' );
+		$this->assertFalse( has_action( 'amp_print_analytics' ) );
+		$this->assertFalse( has_action( 'wp_footer' ) );
+		$this->assertFalse( has_action( 'amp_post_template_footer' ) );
+		$this->assertFalse( has_filter( 'amp_post_template_data' ) );
+
+		$tagmanager->set_data( 'use-snippet', array( 'useSnippet' => true ) );
+		$tagmanager->set_data(
+			'container-id',
+			array(
+				'containerID'  => '9999999',
```

Comment:
`containerID` should be a valid `GTM-` string rather than an integer (unlike `internalContainerID`). I realize it probably doesn't matter for the logic used but we should be consistent in cases where we check with something that should work or not to use valid forms.